web/app/templates/layout.html

```diff
@@ -32,6 +33,17 @@
             <li><a href="{{ url_for('data') }}">Data</a></li>
             <li><a href="{{ url_for('checkpoints') }}">Checkpoints</a></li>
         </ul>
+        <ul class="nav navbar-nav navbar-right">
+            <li class="dropdown">
+                <a href="#" id="currentUser" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false"> <span class="caret"></span></a>
+                <ul class="dropdown-menu" id="userSelect">
+                    {% for id, user in users.items() %}
+                        <li id="select{{id}}"><a value="{{id}}" href="#" class="userSelect">{{user.username}}</a></li>
```

Comment:
Another nitpicky thing, but I think the style with these ninja templates is to put spaces before and after the double curly braces, like so:
```suggestion
                        <li id="select{{ id }}"><a value="{{ id }}" href="#" class="userSelect">{{ user.username }}</a></li>
```